packages/account/src/Sections/Verification/ProofOfOwnership/expanded-card.jsx

```diff
@@ -0,0 +1,200 @@
+    const handleUploadedFile = (name, file) => {
+        setFieldValue(name, file);
+    };
+    const handleBlur = (name, payment_method_identifier, item_id, item_index) => {
+        handleIdentifierChange(
+            name,
+            formatIdentifier(payment_method_identifier, card_details?.identifier_type),
+            item_id,
+            item_index
+        );
+    };
+    const handleIdentifierChange = (name, payment_method_identifier, item_id, item_index) => {
+        setFieldValue(`${name}`, {
+            ...values.data?.[index]?.[item_index],
+            id: item_id,
+            payment_method_identifier,
+            is_generic_pm: card_details.is_generic_pm,
+            identifier_type: card_details.identifier_type,
+        });
+    };
+    const exampleLink = () =>
+        card_details?.identifier_type === IDENTIFIER_TYPES.card_number && (
+            <span
+                className='proof-of-ownership__card-open-desc-link'
+                key={0}
+                onClick={() => {
+                    setIsSampleModalOpen(true);
+                }}
+            >
+                {localize('See example')}
+            </span>
+        );
+    const formatIdentifier = (payment_method_identifier, identifier_type) => {
+        let formatted_id = payment_method_identifier?.replace(/\s/g, '') ?? '';
+        if (identifier_type === IDENTIFIER_TYPES.card_number) {
+            if (
+                formatted_id.length !== 16 ||
+                (formatted_id.length === 16 && VALIDATIONS.has_invalid_characters(formatted_id))
+            ) {
+                return formatted_id;
+            }
+            formatted_id = `${formatted_id.substring(0, 6)}XXXXXX${formatted_id.substring(12)}`;
+        } else if ([IDENTIFIER_TYPES.email_address, IDENTIFIER_TYPES.user_id].some(s => s === identifier_type)) {
+            return formatted_id;
+        }
+        return formatted_id.replace(/(\w{4})/g, '$1 ').trim();
+    };
+    const isSpecialPM = pm_icon => ['IcOnlineNaira', 'IcAstroPayLight', 'IcAstroPayDark'].some(ic => ic === pm_icon);
+    return (
+        <>
+            <div>
+                {card_details?.instructions?.map(instruction => (
+                    <Text
+                        className='proof-of-ownership__card-open-desc'
+                        as='p'
+                        color='general'
+                        size='xs'
+                        key={instruction?.key ?? instruction}
+                    >
+                        {instruction} {exampleLink()}
+                    </Text>
+                ))}
+                <fieldset>
+                    {card_details.items.map((item, item_index) => {
+                        const controls_to_show = [...Array(item?.documents_required).keys()];
+                        return (
+                            <div className='proof-of-ownership__card-open-inputs' key={`${item_index}-${item.id}`}>
```

Comment:
I suppose we shouldn’t be connected with index of element using map, because it is useless for React performance. I think in terms of rerendering performance  this expression  `${item_index}-${item.id}` is almost the same like using just index alone

Reply:
No @yauheni-kryzhyk-deriv, the elements don't have unique ids, so in this case, there is no choice, but to use the index. Also, this is safe because the list is fixed and it doesn't change (it's never rearranged and we never remove or add items to it). This should be fine man.